msaad.go

```diff
@@ -523,6 +525,56 @@ func (m *MSAAD) syncRoles(roleGroups *cachedRoleGroups, aadUser *msaadUser, inte
 	groupsChanged := false
 	userHasAnyIMQSPermission := false
 
+	// identify unmapped groups
+	removeIDs, _ := DecodePermit(make([]byte, 0))
+	allowedIDs, _ := DecodePermit(make([]byte, 0))
+
+	m.parent.Log.Debugf("MSAAD empty role arrays constructed")
+
+	// get all mapped group ids
+	for _, internalGroupName := range m.Config.RoleToGroup {
+		m.parent.Log.Debugf("MSAAD checking all roles: %v", internalGroupName)
+
+		internalGroup, ok := roleGroups.nameToGroup[internalGroupName]
+		if !ok {
+			// We've already logged an error about this, so here we just ignore it
+			continue
+		}
+		m.parent.Log.Debugf("MSAAD add allowed ID for %v", internalGroupName)
+		allowedIDs = append(allowedIDs, internalGroup.ID)
+	}
+
+	for _, groupName := range m.Config.DefaultRoles {
+		m.parent.Log.Debugf("MSAAD checking default roles: %v", groupName)
+
+		internalGroup, ok := roleGroups.nameToGroup[groupName]
+		if !ok {
+			// We've already logged an error about this, so here we just ignore it
+			continue
+		}
+		m.parent.Log.Debugf("MSAAD add allowed default ID for %v", groupName)
+		allowedIDs = append(allowedIDs, internalGroup.ID)
+	}
+
+	// now remove all IDs from groupID that is NOT in allowedIDs
+	for _, groupID := range groupIDs {
+		if allowedIDs.IndexOf(groupID) == -1 {
+			m.parent.Log.Debugf("MSAAD unmapped ID %v, add to remove list", groupID)
+			removeIDs = append(removeIDs, groupID)
+		}
+	}
+
+	for _, id := range removeIDs {
+		if idx := groupIDs.IndexOf(id); idx != -1 {
+			group := roleGroups.idToGroup[id]
+			groupName := group.Name
```

Comment:
Superfluous creation of the variable here. Rather just pass the ltype in the next line.